src/ui/uas/UASMessageView.cc

```diff
@@ -0,0 +1,159 @@
+}
+
+/*-------------------------------------------------------------------------------------
+  UASMessageViewWidget
+-------------------------------------------------------------------------------------*/
+
+UASMessageViewWidget::UASMessageViewWidget(QWidget *parent)
+    : UASMessageView(parent)
+    , _unconnectedWidget(NULL)
+{
+    setStyleSheet("QPlainTextEdit { border: 0px }");
+    // Construct initial widget
+    _unconnectedWidget = new QGCUnconnectedInfoWidget(this);
+    ui()->horizontalLayout->addWidget(_unconnectedWidget);
+    ui()->plainTextEdit->hide();
+    // Enable the right-click menu for the text editor. This works because the plainTextEdit
+    // widget has its context menu policy set to its actions list. So any actions we add
+    // to this widget's action list will be automatically displayed.
+    // We only have the clear action right now.
+    QAction* clearAction = new QAction(tr("Clear Text"), this);
+    connect(clearAction, SIGNAL(triggered()), ui()->plainTextEdit, SLOT(clear()));
+    ui()->plainTextEdit->addAction(clearAction);
+    // Connect message handler
+    connect(UASMessageHandler::instance(), SIGNAL(textMessageReceived(UASMessage*)), this, SLOT(handleTextMessage(UASMessage*)));
+}
+
+UASMessageViewWidget::~UASMessageViewWidget()
+{
+
+}
+
+void UASMessageViewWidget::handleTextMessage(UASMessage *message)
+{
+    // Reset
+    if(!message) {
+        ui()->plainTextEdit->clear();
+        _unconnectedWidget->show();
+        ui()->plainTextEdit->hide();
+    } else {
+        // Make sure the UI is configured for showing messages.
+        // Note that this call is NOT equivalent to `_unconnectedWidget->isVisible()`.
+        if (!_unconnectedWidget->isHidden())
+        {
+            _unconnectedWidget->hide();
+            ui()->plainTextEdit->show();
+        }
+        QPlainTextEdit *msgWidget = ui()->plainTextEdit;
+        // Turn off updates while we're appending content to avoid breaking the autoscroll behavior
+        msgWidget->setUpdatesEnabled(false);
+        QScrollBar *scroller = msgWidget->verticalScrollBar();
+        msgWidget->appendHtml(message->getFormatedText());
+        // Ensure text area scrolls correctly
+        scroller->setValue(scroller->maximum());
+        msgWidget->setUpdatesEnabled(true);
+    }
+}
+
+/*-------------------------------------------------------------------------------------
+  UASMessageViewRollDown
+-------------------------------------------------------------------------------------*/
+
+UASMessageViewRollDown::UASMessageViewRollDown(QWidget *parent, QGCToolBar *toolBar)
+    : UASMessageView(parent)
+{
+    _toolBar = toolBar;
+    setAttribute(Qt::WA_TranslucentBackground);
+    setStyleSheet("background-color: rgba(0%,0%,0%,80%); border: 2px;");
+    QPlainTextEdit *msgWidget = ui()->plainTextEdit;
+    QAction* clearAction = new QAction(tr("Clear Text"), this);
+    connect(clearAction, SIGNAL(triggered()), msgWidget, SLOT(clear()));
```

Comment:
Got it. FYI for now, or try to fix if you want: I've been trying to converting as many of these over as I hit them. There are two ways to connect:

Method 1: connect(signalObject, SIGNAL(...), slotObject, SLOT(...);
Using this syntax things will still compile if SIGNAL or SLOT are missing or incompatible. Instead you'll get a runtime warning.

Method 2: connect(signalObject, &SignalObjectClass::signalMethod, slotObject, &SlotObjectClass::slotMethod);
Using this syntax the compile will fail if either signal or slot are incorrect.

What tends to happen is that later on, someone changes the method for a signal or a slot without fixing all the method 1 type connect calls. This leads to errors that only show up at runtime. If you use method 2 this will automatically break at compile time. It is not always possible to use method 2. For example some signals have multiple overriden methods. It gets nasty in those cases.